internal/internal_task_handlers.go

```diff
@@ -494,15 +493,10 @@ func (w *workflowExecutionContextImpl) Unlock(err error) {
 }
 
 func (w *workflowExecutionContextImpl) getEventHandler() *workflowExecutionEventHandlerImpl {
-	eventHandler := w.eventHandler.Load()
-	if eventHandler == nil {
+	if w.eventHandler == nil {
 		return nil
 	}
-	eventHandlerImpl, ok := eventHandler.(*workflowExecutionEventHandlerImpl)
-	if !ok {
-		panic("unknown type for workflow execution event handler")
-	}
-	return eventHandlerImpl
+	return (*w.eventHandler).(*workflowExecutionEventHandlerImpl)
```

Comment:
do we need to grab a lock for this?

Reply:
since previously this is stored in atomic.Value

Reply:
Putting it here can cause deadlocks - generally functions calling into this one are doing the locking. Overall there's not much consistency around when the lock is taken. I'd be worth cleaning this up as part of the bigger refactorings I'll take on soon.